pixels/utils/ratelimits.py

```diff
@@ -305,3 +316,11 @@ async def _get_remaining_cooldown(self, request_id: int) -> int:
         key = f"cooldown-{self.ROUTE_NAME}-{self.state[request_id].user_id}"
 
         return await self.redis.ttl(key)
+
+    async def _reset_time(self, request_id: int) -> int:
+        key = f"interaction-{self.ROUTE_NAME}-{self.state[request_id].user_id}"
+
+        if not (newest_uuid := await self.redis.zrange(key, 0, 0)):
+            return -1
+
+        return await self.redis.zscore(key, newest_uuid[0]) - time()
```

Comment:
Should we wrap this in a `math.ceil()` call? That way the header returns a nice round int, rather than a float with a load of decimal places

Reply:
I think decimal places have their value here as it allows to have more precise remaining time that has something rounded up. We can truncate after a few decimal places if you wish though.